back/app/services/survey_results_generator_service.rb

```diff
@@ -24,23 +24,31 @@ def generate_results
   end
 
   def visit_select(field)
+    option_keys = field.options.each_with_object({}) do |option, accu|
```

Comment:
We can simplify this code:

```suggestion
    option_keys = field.options.pluck(:key)
```